src/models/model.cpp

```diff
@@ -1078,13 +1962,16 @@ void model::update_weights() {
   // after a weights gradient has been computed. Thus, iterating in
   // reverse order will use gradients that have already finished their
   // allreduce, giving more time for more recent allreduces to finish.
+  int counter = 0;
   for (auto rit = m_weights.rbegin(); rit != m_weights.rend(); ++rit) {
     auto& w = **rit;
     auto&& opt = w.get_optimizer();
+
     if (opt != nullptr) {
       do_weight_optimize_begin_cbs(&w);
       opt->step();
       do_weight_optimize_end_cbs(&w);
+      counter++;
```

Comment:
`counter` doesn't do anything:
```suggestion
  for (auto rit = m_weights.rbegin(); rit != m_weights.rend(); ++rit) {
    auto& w = **rit;
    auto&& opt = w.get_optimizer();
    if (opt != nullptr) {
      do_weight_optimize_begin_cbs(&w);
      opt->step();
      do_weight_optimize_end_cbs(&w);
```